src/core/dbus-unit.c

```diff
@@ -722,6 +723,53 @@ int bus_unit_method_clean(sd_bus_message *message, void *userdata, sd_bus_error
         return sd_bus_reply_method_return(message, NULL);
 }
 
+static int bus_unit_method_freezer_generic(sd_bus_message *message, void *userdata, sd_bus_error *error, FreezerAction action) {
+        const char* perm;
+        int (*method)(Unit*);
+        Unit *u = userdata;
+        int r;
+
+        assert(message);
+        assert(u);
+        assert_return(IN_SET(action, FREEZER_FREEZE, FREEZER_THAW), -EINVAL);
+
+        if (action == FREEZER_FREEZE) {
+                perm = "stop";
+                method = unit_freeze;
+        } else {
+                perm = "start";
+                method = unit_thaw;
+        }
+
+        r = mac_selinux_unit_access_check(u, message, perm, error);
+        if (r < 0)
+                return r;
+
+        r = sd_bus_message_new_method_return(message, &u->pending_freezer_message);
```

Comment:
hmm, we blindly override `pending_freezer_message` here. We should refuse operation if the field is already initialized, i.e. if there's already a freeze operation pending.